### PR TITLE
Make sure .all() call is covered

### PR DESCRIPTION
**Pull request by marcofucci**
When Clazz.objects.all() gets called:
- cdms objs that don't exist in local, get created locally
- local objs out-of-date get updated from cdms ones

In case of exceptions during cdms calls, the error is propagated.

As cdms only supports 50 window size results and it's not efficient
to iterate over all the list every time, the developer has to acknowledge that
the .all() call might not include all the objects.
This is unfortunately a limitation imposed by the tech constraints.

Clazz.objects.skip_cdms().all() will skip cdms calls and only hit the local db.

### CODE REVIEW

**Comment by marcofucci** [comment]
@danielquinn code review? What do you think about cdms exceptions propagating instead of failing silently?

File: crm-poc/apps/cdms_api/tests/utils.py [hunk 1]
```diff
@@ -6,40 +8,41 @@
 from cdms_api.utils import datetime_to_cdms_datetime
 
 
-def mocked_cdms_get(modified_on=None, get_data={}):
+def populate_data(data, service, guid=None):
```
Comment (danielquinn) [inline crm-poc/apps/cdms_api/tests/utils.py]: This needs a docstring explanation and/or a name change.  "populate data" is just too generic, especially when it's in a file called "utils.py"

**Comment by danielquinn** [comment]
In terms of exceptions bubbling vs. being made silent, I usually like to catch them and re-throw them as new exceptions to keep the outside ignorant of what's going on under the hood.  For example:

```python

from magic.data.store import db, DataStoreException

class DoesNotExist(Exception):
    pass

class MyModel(object):
    def get_stuff(self):
        try:
            return db.get(pk=pk)
        except DataStoreException:
            raise DoesNotExist("I have no idea what you're talking about")
```

This prevents you from silencing exceptions that shouldn't be, but doesn't spread the exceptions around a project that shouldn't need to know about that specific type of exception.

Of course you can always silence exceptions if they really are no danger, but then obviously we have to be very specific about which exceptions we're ignoring.  I've seen too many blocks of code that look like this:

```python
    try:
        # Something
    except:
        pass
```

**Comment by marcofucci** [comment]
That makes sense for exceptions that a dev could expect to find (e.g. `DoesNotExist`) as the meaning of it if clear even when ignoring what's going on under the hood.

However, for unexpected exceptions like these, the dev does not expect a call to `.all()` to fail and re-throwing the exception would actually be worse as it would make it even less obvious what the actual problem is.
Don't you think?

**Pull request merged**